dataprofiler/tests/profilers/profiler_options/test_unstructured_options.py

```diff
@@ -148,3 +151,20 @@ def test_eq(self):
         self.assertNotEqual(options, options2)
         options2.text.stop_words = ["woah", "stop", "right", "there"]
         self.assertEqual(options, options2)
+
+    def test_json_encode(self):
+        option = UnstructuredOptions()
+
+        serialized = json.dumps(option, cls=ProfileEncoder)
+
+        expected_class = "UnstructuredOptions"
+        expected_options_attributes = {"text", "data_labeler"}
+
+        actual_option_json = json.loads(serialized)
+
+        self.assertIn("class", actual_option_json)
+        self.assertEqual(expected_class, actual_option_json["class"])
+        self.assertIn("data", actual_option_json)
+        self.assertEqual(
+            expected_options_attributes, set(actual_option_json["data"].keys())
+        )
```

Comment:
I just tested a theory that actually cleans up the code quite a bit. Not sure if it is better than just the deserial we were going to do, but:
```python
def test_json_encode(self):
        option = UnstructuredOptions()

        serialized = json.dumps(option, cls=ProfileEncoder)

        expected_class = "UnstructuredOptions"
        expected_options_attributes = {"text", "data_labeler"}
        expected_json = {
            "class": "UnstructuredOptions",
            "data": {
                "text": {"class": "TextProfilerOptions", "data": mock.ANY},
                "data_labeler": {"class": "DataLabelerOptions", "data": mock.ANY},
            }
        }

        actual_option_json = json.loads(serialized)
        self.assertDictEqual(expected_json, actual_option_json)
```

Reply:
the `mock.ANY` will match anything in that, which we assume is correct in our cases bc of other tests.

Reply:
done